### PR TITLE
feat(spans): Handle span schema v1 from Relay

### PR DESCRIPTION
**Pull request by phacops**
Related to https://github.com/getsentry/relay/pull/2556

### CODE REVIEW

File: src/sentry/spans/consumers/process/factory.py [hunk 1]
```diff
@@ -158,12 +163,58 @@ def _format_event_id(payload: Mapping[str, Any]) -> str:
...
-    relay_span["project_id"] = payload["project_id"]
-    relay_span["event_id"] = _format_event_id(payload)
-    snuba_span = _build_snuba_span(relay_span)
+    if payload.get("version") == SPAN_SCHEMA_V1:
+        snuba_span = _process_relay_span_v1(payload)
+    else:
+        relay_span = payload["span"]
+        relay_span["project_id"] = payload["project_id"]
+        relay_span["event_id"] = _format_event_id(payload)
+        snuba_span = _process_relay_span_v0(relay_span)
```
Comment (nikhars) [inline src/sentry/spans/consumers/process/factory.py:217]: Is my understanding correct that the 2 versions are needed to perform the switchover in a seamless manner. Once the switchover is done, the older version can be gotten rid of?

**Review by nikhars — APPROVED** [review]
Overall looks good to me. Left a comment for clarification.

File: src/sentry/spans/consumers/process/factory.py (hunk 1 above)
Comment (phacops) [reply src/sentry/spans/consumers/process/factory.py:217]: Yes, that's correct.

File: src/sentry/spans/consumers/process/factory.py [hunk 2]
```diff
@@ -158,12 +155,34 @@ def _format_event_id(payload: Mapping[str, Any]) -> str:
...
+    payload = _deserialize_payload(message.payload.value)
     relay_span = payload["span"]
     relay_span["project_id"] = payload["project_id"]
     relay_span["event_id"] = _format_event_id(payload)
-    snuba_span = _build_snuba_span(relay_span)
+
+    organization_id = payload.get("organization_id")
+    retention_days = payload.get("retention_days")
+
+    if not (organization_id and retention_days):
```
Comment (Zylphrex) [inline src/sentry/spans/consumers/process/factory.py:174]: Nit: may be a personal preference but `if not organization_id or not retention_days` reads clearer than this shortened version.

File: src/sentry/spans/consumers/process/factory.py [hunk 3]
```diff
@@ -158,12 +155,34 @@ def _format_event_id(payload: Mapping[str, Any]) -> str:
     return ""
 
 
+def _deserialize_payload(payload: bytes) -> Mapping[str, Any]:
+    try:
+        return msgpack.unpackb(payload)
+    except msgpack.FormatError:
+        return json.loads(payload, use_rapid_json=True)
```
Comment (Zylphrex) [inline src/sentry/spans/consumers/process/factory.py:162]: Can we add a comment explaining this is for backwards compatibility between v1 and v2?

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**